packages/datetime/src/dateInput.tsx

```diff
@@ -165,24 +148,17 @@ export class DateInput extends AbstractPureComponent<IDateInputProps, IDateInput
 
     public static displayName = "Blueprint2.DateInput";
 
-    public constructor(props?: IDateInputProps, context?: any) {
-        super(props, context);
-
-        const defaultValue = this.props.defaultValue ? fromDateToMoment(this.props.defaultValue) : moment(null);
-
-        this.state = {
-            isInputFocused: false,
-            isOpen: false,
-            value: this.props.value !== undefined ? fromDateToMoment(this.props.value) : defaultValue,
-            valueString: null,
-        };
-    }
+    public state: IDateInputState = {
```

Comment:
are you sure it's ok to move this to the static member? why did we have it in the constructor before?

Reply:
i am quite sure. typescript always inline state init in the constructor that it generates (before all the other `=>` bound methods), so you can actually safely use `this.props` in `public state` initialization.